apps/generator/lib/conditionalGeneration.js

```diff
@@ -0,0 +1,269 @@
+ * @param {String} matchedConditionPath The matched condition path.
+ * @param {String} relativeSourceDirectory The relative path of the source directory.
+ * @param {String} relativeTargetFile The relative path of the target file.
+ * @param {String} targetDir The directory where the generated files are written.
+ * @return {Promise<Boolean>} A promise that resolves to true if the generation should be skipped.
+ */
+async function validateParameterValue(
+  validation,
+  argument,
+  matchedConditionPath,
+  relativeSourceDirectory,
+  relativeTargetFile,
+  targetDir
+) {
+  if (Object.hasOwn(validation, 'not')) {
+    const isNotValid = validateNot(validation.not, argument);
+    if (isNotValid && (matchedConditionPath === relativeSourceDirectory)) {
+      await removeParentDirectory(relativeTargetFile, targetDir);
+      return false;
+    } else 
+    if (isNotValid && (matchedConditionPath === relativeTargetFile)) {
+      return false;
+    }
+    log.debug(logMessage.conditionalGenerationMatched(matchedConditionPath));
+    return true;
+  }
+
+  const ajv = new Ajv();
+  const validate = ajv.compile(validation);
+  const isValid = validate(argument);
+
+  if (!isValid) {
+    log.debug(logMessage.conditionalGenerationMatched(matchedConditionPath));
+    return true;
+  } else 
+  if (isValid && (matchedConditionPath === relativeTargetFile)) {
+    return false;
+  } else
+  if (isValid && (matchedConditionPath === relativeSourceDirectory)) {
+    await removeParentDirectory(relativeTargetFile, targetDir);
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Validates the parameter value using a "not" schema.
+ *
+ * @param {Object} notSchema The "not" schema to use for validation.
+ * @param {any} parameterValue The value to validate.
+ * @return {Boolean} Returns true if validation fails, false otherwise.
+ */
+function validateNot(notSchema, parameterValue) {
+  const ajv = new Ajv();
+  const validateNot = ajv.compile(notSchema);
+  return validateNot(parameterValue);
+}
+
+/**
+ * Removes the parent directory of the specified target file if it exists and is a directory.
+ *
+ * @param {String} relativeTargetFile The relative path of the target file used to determine the parent directory.
+ * @param {String} targetDir The directory where the generated files are written.
+ * @return {Promise<void>} A promise that resolves once the parent directory is removed (if applicable).
+ */
+function removeParentDirectory(relativeTargetFile, targetDir) {
+  const fullFilePath = path.join(targetDir, relativeTargetFile);
+  const parentDir = path.dirname(fullFilePath);
+
+  if (fs.existsSync(parentDir)) {
+    const stats = fs.lstatSync(parentDir);
+    if (stats.isDirectory()) {
+      fs.rmdirSync(parentDir, { recursive: true });
+    }
+  }
+}
+
+/**
+ * Retrieves the parameter value from the AsyncAPI document.
+ *
+ * @param {AsyncAPIDocument} asyncapiDocument The AsyncAPI document to extract the parameter value from.
```

Comment:
do not match the name of the argument in function

Reply:
Changed thanks @derberg